### PR TITLE
Fix bug in _from_hgvs when using SeqRepoDataProxy

### PR DESCRIPTION
**Pull request by korikuzma**
~Same PR as #56~ The following was copied from #56:

`_from_hgvs` works when using SeqRepoRestDataProxy, but failed when using SeqRepoDataProxy for identity and dup HGVS expressions. This failed because `fetch_uri` expects a namespace and accession (ex: refseq:NM_12345) and `sv.ac` would only provide an accession (ex: NM_12345), so switched to using `sequence_id` as the identifier. 

> def fetch_uri(self, uri, start=None, end=None):
        """fetch sequence for URI/CURIE of the form namespace:alias, such as
        NCBI:NM_000059.3.
>   
>        """
>    
>       namespace, alias = uri_re.match(uri).groups()
>       AttributeError: 'NoneType' object has no attribute 'groups'

Added tests for both SeqRepoRestDataProxy and SeqRepoDataProxy. 

Notes:
- I generated updated versions of the VCR cassettes
- I removed ambiguous variants from `tests/test_vrs_normalize.py` since we do not know how to normalize these yet
- CI/CD is only failing on Python 3.9 due to issues with installing pysam

### CODE REVIEW

**Review by larrybabb — COMMENTED** [review]
It looks fine to me. But I will certainly defer to the python folks for final word. @korikuzma can we start a thread on slack about the issue with handling ambiguous variants? I'd like to understand precisely what the issue is and if we can provide a plan to deal with these cases.

File: tests/extras/test_translator.py [hunk 1]
```diff
@@ -25,23 +25,31 @@
 }
 
 
+@pytest.mark.parametrize("tlr_fixture_name", ("rest_tlr","tlr"))
```
Comment (reece) [inline tests/extras/test_translator.py:28]: I assume you tried to pass `rest_tlr` and `tlr` (unquoted) in the parametrize setup, right? I can see what's happening, but it's unclear to me why this indirection is necessary. (It's fundamentally the same as passing functions to any other function, I think.)

So, please just confirm that the lookup/indirection is needed. If it's needed, then fine (and nice workaround!).

**Review by reece — CHANGES_REQUESTED** [review]
@korikuzma: Exceptional work! Thanks for addressing this issue.

I made several comments inline.

Now a broader question: Can you please comment on why you think vrs-python should be testing both local and rest seqrepo dataproxies? IMO, the bug here is that the data proxies have slightly different behavior. We should fix that. A different approach is to test that the data proxies have identical behavior for lookups (e.g., with fully-qualified and unqualified identifiers), then test each normalization only once. I think this would be preferable for these reasons: 1) It tests the underlying cause of the bug, 2) If we add another dp based on, say, Redis, we now have to run three times; and 3) With this change, we're doing twice as much work as needed to test normalization. If a bug appears in normalization, we'll now see it twice.

File: tests/extras/test_translator.py (hunk 1 above)
Comment (korikuzma) [reply tests/extras/test_translator.py:28]: Correct. Quotes are necessary due to [request.getfixturevalue](https://docs.pytest.org/en/6.2.x/reference.html#pytest.FixtureRequest.getfixturevalue), whose argument is a string, on the following lines.

File: tests/test_vrs_normalize.py [hunk 2]
```diff
@@ -46,97 +46,33 @@
     "type": "Allele"
 }
 
-allele_dict2 = {
```
[the remark this answers is not in the document]
Comment (korikuzma) [reply tests/test_vrs_normalize.py:49]: @ahwagner 's comment in the slack thread:
> The idea here is that if you have an Allele with an ambiguous location (coordinates defined by definite or indefinite ranges), the notion of normalizing (e.g. by VOCA) is not tenable.

**Comment by korikuzma** [comment]
@reece I tested both local and rest seqrepo dataproxies since I was getting an error in `_from_hgvs` when using the local but not the rest dataproxy. So I wanted to ensure that both give the same end result during tests. 

I agree with your recommendation. Should we make a separate issue to address this?

File: tests/test_vrs_normalize.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (reece) [reply tests/test_vrs_normalize.py:49]: Right. We should test that normalize does *not* modify the Allele.

[the remark this answers is not in the document]
Comment (korikuzma) [reply tests/test_vrs_normalize.py:49]: Sounds good. Let me know if this should be addressed in this issue or if it's okay to create a new issue for this.

**Comment by reece** [comment]
Here's my recommendation:

* Let's close this PR unmerged.
* Update GenomicMedLab/vrs-python to current vrs-python. (The seqrepo-dp branch was based on main from ~4 months ago, which might be the cause of merge problems. See below.)
* Create a new issue with the observation that the dp returns different data when qualified or not.
* Create a branch like 999-seqrepo-dp (where 999 is the issue number above)
* Use a parametrize setup similar to what you created here to validate the seqrepo replies.
* Delete this branch.

How does that sound?

image

**Comment by korikuzma** [comment]
@reece Sounds good! I had merged `ga4gh/vrs-python` main branch into `GenomicMedLab/vrs-python` seqrepo-dp branch before creating the PR. I think the merge issues are caused by the python 3.9 CI/CD not passing due to the pysam dependency issue.

**Pull request closed without merging**